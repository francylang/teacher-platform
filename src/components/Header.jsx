import React from 'react';
import { Link } from 'react-router-dom';

const Header = ({ signedInStatus }) => {
  const renderSignOut = () => {
    if (signedInStatus === true) {
      return (
        <h3 className="sign-out">Sign Out</h3>
      );
    }
  };

  return (
    <header className="header">
      <div className="header-icon-container">
        <Link to='/' className="header-link"><h1 className="app-name">Edvice</h1></Link>
        <div className="header-image"></div>
      </div>
      {renderSignOut()}
    </header>
  );
};

export default Header;
