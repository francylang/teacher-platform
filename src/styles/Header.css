@import url("https://fonts.googleapis.com/css?family=Ubuntu+Mono");
.header {
  background: #5c258d;
  background: -webkit-linear-gradient(to right, #5c258d, #4389a2);
  background: linear-gradient(to right, #5c258d, #4389a2);
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 15px;
  font-family: 'Ubuntu Mono'; }

.app-name {
  color: #fff;
  margin: 0; }
