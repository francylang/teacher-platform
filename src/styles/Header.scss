.header {
  background-color: #F4F8FB;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 15px 2%;
}

.app-name {
  color: #2B2B2B;
  margin: 0;
  font-family: 'Ubuntu Mono';
}

.sign-out {
  color: #2B2B2B;
  margin: 0;
  font-family: 'Ubuntu Mono';
}

.sign-out:hover {
  cursor: pointer;
}
