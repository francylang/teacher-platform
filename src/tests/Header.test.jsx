import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../components/Header';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<Header />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a header', () => {
    const header = wrapper.find('header');

    expect(header.length).toEqual(1);
  });

  it('should render an app name', () => {
    const name = wrapper.find('.app-name');

    expect(name.length).toEqual(1);
  });

  it('should render a sign out link', () => {
    const signOut = wrapper.find('.sign-out');

    expect(signOut.length).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
