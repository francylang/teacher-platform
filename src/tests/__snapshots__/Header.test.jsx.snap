// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-icon-container"
        >
          <Link
            className="header-link"
            replace={false}
            to="/"
          >
            <h1
              className="app-name"
            >
              Edvice
            </h1>
          </Link>
          <div
            className="header-image"
          />
        </div>,
        undefined,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="header-link"
              replace={false}
              to="/"
            >
              <h1
                className="app-name"
              >
                Edvice
              </h1>
            </Link>,
            <div
              className="header-image"
            />,
          ],
          "className": "header-icon-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h1
                className="app-name"
              >
                Edvice
              </h1>,
              "className": "header-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edvice",
                "className": "app-name",
              },
              "ref": null,
              "rendered": "Edvice",
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-image",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-icon-container"
          >
            <Link
              className="header-link"
              replace={false}
              to="/"
            >
              <h1
                className="app-name"
              >
                Edvice
              </h1>
            </Link>
            <div
              className="header-image"
            />
          </div>,
          undefined,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="header-link"
                replace={false}
                to="/"
              >
                <h1
                  className="app-name"
                >
                  Edvice
                </h1>
              </Link>,
              <div
                className="header-image"
              />,
            ],
            "className": "header-icon-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h1
                  className="app-name"
                >
                  Edvice
                </h1>,
                "className": "header-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edvice",
                  "className": "app-name",
                },
                "ref": null,
                "rendered": "Edvice",
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "header-image",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
